amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/common/EPConstants.java

```diff
@@ -112,4 +112,10 @@
 
     public static final String ISO8601_DATE_AND_TIME_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSZ";
     public static final String ISO8601_DATE_FORMAT = "yyyy-MM-dd";
+    
+    public static final Map<String, Integer> DATE_FORMAT_STRICT_LENGTH = new HashMap<String, Integer>() {{
+        put(ISO8601_DATE_AND_TIME_FORMAT, 28);
+        put(ISO8601_DATE_FORMAT, 10);
```

Comment:
For clarity you could use take the length based on a sample like `"2019-02-26T11:08:56.235-0200".lenght()` :) for timestamp and `ISO8601_DATE_FORMAT.length()` for date

Reply:
For clarity I will add comments :)